src/test/java/com/exactpro/epfast/decoder/ascii/TestDecodeAsciiString.java

```diff
@@ -10,164 +12,199 @@
-        withByteBuf("00 00 80", buffers -> {
-            decode(mandatoryStringDecoder, buffers);
-            assertTrue(mandatoryStringDecoder.isReady());
-            assertEquals("\0\0", mandatoryStringDecoder.getValue());
-        });
-    }
-
-    @Test
-    void testZeroByteStringMandatory2() throws IOException {
-        withByteBuf("00 00 00 00 80", buffers -> {
-            decode(mandatoryStringDecoder, buffers);
-            assertTrue(mandatoryStringDecoder.isReady());
-            assertEquals("\0\0\0\0", mandatoryStringDecoder.getValue());
-        });
-    }
-
-    @Test
-    void testOverlong1() throws IOException {
-        withByteBuf("00 81", buffers -> {
-            decode(mandatoryStringDecoder, buffers);
-            assertTrue(mandatoryStringDecoder.isReady());
-            mandatoryStringDecoder.setCheckOverlong();
-
-            assertThrows(IOException.class, () -> mandatoryStringDecoder.getValue());
-            mandatoryStringDecoder.clearCheckOverlong();
-            mandatoryStringDecoder.getValue();
-        });
-    }
-
-    @Test
-    void testOverlong2() throws IOException {
-        withByteBuf("00 00 00 81", buffers -> {
-            decode(nullableStringDecoder, buffers);
-            assertTrue(nullableStringDecoder.isReady());
-            nullableStringDecoder.setCheckOverlong();
-
-            assertThrows(IOException.class, () -> nullableStringDecoder.getValue());
-            nullableStringDecoder.clearCheckOverlong();
-            nullableStringDecoder.getValue();
-        });
-    }
-
-    @Test
-    void testNullableReuse() throws IOException {
-        withByteBuf("41 42 c3 42 42 c3 41 44 c3", buffers -> {
-            decode(nullableStringDecoder, buffers);
-            assertTrue(nullableStringDecoder.isReady());
-            assertEquals("ABC", nullableStringDecoder.getValue());
-            decode(nullableStringDecoder, buffers);
-            assertEquals("BBC", nullableStringDecoder.getValue());
-            decode(nullableStringDecoder, buffers);
-            assertEquals("ADC", nullableStringDecoder.getValue());
-        });
-    }
-
-    @Test
-    void testSimpleStringGetValueTwice() throws IOException {
-        withByteBuf("41 42 c3", buffers -> {
-            decode(nullableStringDecoder, buffers);
-            assertTrue(nullableStringDecoder.isReady());
-            assertEquals("ABC", nullableStringDecoder.getValue());
-            assertEquals("ABC", nullableStringDecoder.getValue());
-        });
-    }
-
-    @Test
-    void testMandatoryEmptyStringGetValueTwice() throws IOException {
-        withByteBuf("80", buffers -> {
-            decode(mandatoryStringDecoder, buffers);
-            assertTrue(mandatoryStringDecoder.isReady());
-            assertEquals("", mandatoryStringDecoder.getValue());
-            assertEquals("", mandatoryStringDecoder.getValue());
-        });
+    static String setLength(int length) {
```

Comment:
chnage to `fastAsciiStringOf(char character, int length)`

add check that character > 0 and < 128 to prevent invalid usage